### PR TITLE
New event detection tests

### PR DESCRIPTION
**Pull request by kanav99**
Repeat tests will fail because of a possible bug in the edge case of callback interpolation logic. I will make a PR on DiffEqBase, but don't merge that now, callback code is hard to understand and I want to be double sure

### CODE REVIEW

File: test/integrators/rev_events_tests.jl [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    f = (u, p, t) -> u
+    cb = ContinuousCallback((u,t,int) -> u[1] - exp(1), terminate!)
+    prob = ODEProblem(f, SVector(5.0), (2.0, 0.0), callback=cb)
+    sol = solve(prob, alg; kwargs...)
+    sol.u[end][1] ≈ exp(1)
+end
+
+function test_callback_mvector(alg; kwargs...)
+    f = (u, p, t) -> copy(u)
+    cb = ContinuousCallback((u,t,int) -> u[1] - exp(1), terminate!)
```
[the remark this answers is not in the document]
Comment (kanav99) [reply test/integrators/rev_events_tests.jl]: I just added the same old tests with reversed time direction. I can add on vector continous callback next.

**Comment by kanav99** [comment]
VectorContinuousCallbacks don't do well with `idxs`

**Comment by ChrisRackauckas** [comment]
That's fine for now.

**Comment by kanav99** [comment]
Then it's good to go, just waiting for the tests. I will be implementing DAE support now in the current NLsolve implementation, while we get a new cleaner NLSolve later

**Comment by kanav99** [comment]
Should we merge now?
